gocd/gocd.go

```diff
@@ -79,6 +79,29 @@ type Pipeline struct {
 	Timer        interface{} `json:"timer"`
 }
 
+// Server a representstion of a GoCD server
+type Server struct {
+	GOCDHost     string
+	GOCDPort     string
+	GOCDUser     string
+	GOCDPassword string
+}
+
+// NewServerConfig Create a Server object from a config
+func NewServerConfig(config *config.Config) *Server {
```

Comment:
As discussed, this should take the individual arguments instead of a config object to reduce coupling.